geomesa-core/src/test/scala/org/locationtech/geomesa/core/index/IndexValueEncoderTest.scala

```diff
@@ -88,10 +89,10 @@ class IndexValueEncoderTest extends Specification {
 
       // requirements
       decoded must not beNull;
-      decoded.attributes must haveSize(3)
-      decoded.id mustEqual id
-      decoded.geom mustEqual geom
-      decoded.date mustEqual Some(dt)
+      decoded.getAttributeCount mustEqual(2)
```

Comment:
Yeah, previously it was a map of key values (including the ID, as you guessed), which we later turned into a simple feature. Now it decodes directly into a simple feature, which lets us standardize the interface and use kryo.